internal/pkg/metrics/controller_metrics.go

```diff
@@ -1,71 +1,127 @@
+				Name: "thanos_operator_paused",
+				Help: "Paused state of ThanosOperator",
+			}, []string{"component", "resource", "namespace"}),
+			ResourceSync: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+				Name: "thanos_operator_resource_sync_total",
+				Help: "Total number of resources synced by ThanosOperator",
+			}, []string{"component", "resource", "namespace", "case"}),
+		}
+	})
+	return commonMetricsInstance
+}
+
+func NewThanosQueryMetrics(reg prometheus.Registerer, commonMetrics *CommonMetrics) ThanosQueryMetrics {
 	return ThanosQueryMetrics{
+		CommonMetrics: commonMetrics,
 		EndpointsConfigured: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
 			Name: "thanos_operator_query_endpoints_configured",
 			Help: "Number of configured endpoints for ThanosQuery resources",
 		}, []string{"type", "resource", "namespace"}),
-		ServiceWatchesReconciliationsTotal: promauto.With(reg).NewCounter(prometheus.CounterOpts{
+		ServiceWatchesReconciliationsTotal: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
 			Name: "thanos_operator_query_service_event_reconciliations_total",
 			Help: "Total number of reconciliations for ThanosQuery resources due to Service events",
-		}),
-		FrontendServiceWatchesReconciliationsTotal: promauto.With(reg).NewCounter(prometheus.CounterOpts{
-			Name: "thanos_operator_query_frontend_service_event_reconciliations_total",
-			Help: "Total number of reconciliations for ThanosQueryFrontend resources due to Service events",
-		}),
+		}, []string{"resource", "namespace"}),
 	}
 }
 
-func NewThanosReceiveMetrics(reg prometheus.Registerer) ThanosReceiveMetrics {
+func NewThanosReceiveMetrics(reg prometheus.Registerer, commonMetrics *CommonMetrics) ThanosReceiveMetrics {
 	return ThanosReceiveMetrics{
+		CommonMetrics: commonMetrics,
 		HashringsConfigured: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
 			Name: "thanos_operator_receive_hashrings_configured",
 			Help: "Number of configured hashrings for ThanosReceive resources",
 		}, []string{"resource", "namespace"}),
 		HashringHash: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
 			Name: "thanos_operator_receive_hashring_hash",
-			Help: "Hash of the hashring for ThanosReceive resources",
+			Help: "Hash of the hashrings configuration for ThanosReceive resources",
 		}, []string{"resource", "namespace"}),
-		EndpointWatchesReconciliationsTotal: promauto.With(reg).NewCounter(prometheus.CounterOpts{
+		HashringTenantsConfigured: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+			Name: "thanos_operator_receive_hashring_tenants_configured",
+			Help: "Number of configured tenants for hashrings for ThanosReceive resources",
+		}, []string{"resource", "namespace", "hashring"}),
+		HashringEndpointsConfigured: promauto.With(reg).NewGaugeVec(prometheus.GaugeOpts{
+			Name: "thanos_operator_receive_hashring_endpoints_configured",
+			Help: "Number of configured endpoints for hashrings for ThanosReceive resources",
```

Comment:
```suggestion
			Help: "Number of configured endpoints for each ThanosReceive hashring",
```

weird help text phrasing no? Or is this some idiom driving this?

Reply:
Yeah just me being lazy, fixed